### PR TITLE
Azure Data Lake - CAMEL-19152 additional file client header field + CAMEL-18636 additional auth options

### PR DESCRIPTION
**Pull request by ank19**
# Description

- see CAMEL-19152 for a detailed description
- patched validateConfiguration() method in the component to accept string based configuration of auth parameters in addition to Java objects
- Added SAS token-based authentication as well as the usage of the default identity if enabled (disabled by default, prioritised if enabled)

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

File: components/camel-azure/camel-azure-storage-datalake/src/main/java/org/apache/camel/component/azure/storage/datalake/DataLakeConfiguration.java [hunk 1]
```diff
@@ -93,6 +94,12 @@ public class DataLakeConfiguration implements Cloneable {
     private String umask;
     @UriParam(description = "set open options for creating file")
     private Set<OpenOption> openOptions;
+    @UriParam(description = "SAS token signature")
```
Comment (davsclaus) [inline components/camel-azure/camel-azure-storage-datalake/src/main/java/org/apache/camel/component/azure/storage/datalake/DataLakeConfiguration.java:97]: Can you add label = "security" for these new options so we know they are related to security stuff

**Comment by oscerd** [comment]
Can you please rebase? @ank19 

Can you please also share how are you setting up your data lake instance? Thanks.

**Comment by oscerd** [comment]
@ank19 this has conflicts because I completely removed the validateConfiguration. To me it's better to notice errors than having to deal with validation.

**Comment by ank19** [comment]
@oscerd - Sure, for me it's fine to drop the validation method, I'm just not sure what is "common" or "usual" in the project. I'm removing it and updating the PR accordingly.

Concerning the Data Lake setup, what I did is basically creating a standard Azure storage account, then, on the overview page, setting Hierarchical namespace to "enabled", which starts that sort of a migration process to Data Lake Gen2. Then I tried three authentication methods for creation of a new file and appending data to that file, specifying an account name and:
1.  Within "Security + networking / Shared access signature": Generate a new SAS with the appropriate permissions and constraints set, then using the "Signature" to set the "sasSignature" as configuration parameter on the component (the thing starting with "?...").
2. Within "Security + networking / Access keys": Using one of the two provided tokens to set "accountKey".
3. Creating an app registration on Azure AD, and then under "Certificates & secrets", creating a new client secret, using the "Value" from that page as "clientSecret", and from the overview page the "Application (client) id" as "clientId" as well as "Directory (tenant) id" as "tenantId". Afterwards, I configured on the storage account, in "Access Control (IAM)", that the app registration just created gets the role "Contributor" for instance. Afterwards I used the Microsoft Storage Explorer to "Propagate Access Control Lists..." on a newly created container, so that the ACLs get applied.

I created a Zulip account - If you like you can try to contact me there, too. However, I might be online from time to time only.

**Comment by ank19** [comment]
@oscerd - You're welcome. Concerning the rebase: I removed the validation method, rebased the commit and run format and the source check; I even tried to run the same check style command as in the action above, but it still failing. Do you have an idea why that is the case?

**Comment by oscerd** [comment]
I think it's something not related to your PR. Once the build will end I'll access the log and let you know.

**Comment by oscerd** [comment]
Not your fault:
```

[ERROR] Failed to execute goal on project camel-dhis2: Could not resolve dependencies for project org.apache.camel:camel-dhis2:jar:3.21.0-SNAPSHOT: Could not find artifact org.apache.camel:camel-dhis2-api:jar:3.21.0-SNAPSHOT in apache.snapshots (https://repository.apache.org/snapshots/) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project camel-dhis2: Could not resolve dependencies for project org.apache.camel:camel-dhis2:jar:3.21.0-SNAPSHOT: Could not find artifact org.apache.camel:camel-dhis2-api:jar:3.21.0-SNAPSHOT in apache.snapshots (https://repository.apache.org/snapshots/)
	at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies(LifecycleDependencyResolver.java:269)
	at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies(LifecycleDependencyResolver.java:147)
	at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved(MojoExecutor.java:401)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:347)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.mvndaemon.mvnd.builder.SmartBuilderImpl.buildProject(SmartBuilderImpl.java:178)
	at org.mvndaemon.mvnd.builder.SmartBuilderImpl$ProjectBuildTask.run(SmartBuilderImpl.java:198)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.maven.project.DependencyResolutionException: Could not resolve dependencies for project org.apache.camel:camel-dhis2:jar:3.21.0-SNAPSHOT: Could not find artifact org.apache.camel:camel-dhis2-api:jar:3.21.0-SNAPSHOT in apache.snapshots (https://repository.apache.org/snapshots/)
	at org.apache.maven.project.DefaultProjectDependenciesResolver.resolve(DefaultProjectDependenciesResolver.java:214)
	at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies(LifecycleDependencyResolver.java:243)
	... 14 common frames omitted
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Could not find artifact org.apache.camel:camel-dhis2-api:jar:3.21.0-SNAPSHOT in apache.snapshots (https://repository.apache.org/snapshots/)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:374)
	at org.apache.maven.project.DefaultProjectDependenciesResolver.resolve(DefaultProjectDependenciesResolver.java:207)
	... 15 common frames omitted
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact org.apache.camel:camel-dhis2-api:jar:3.21.0-SNAPSHOT in apache.snapshots (https://repository.apache.org/snapshots/)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:431)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:235)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:357)
	... 16 common frames omitted
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.apache.camel:camel-dhis2-api:jar:3.21.0-SNAPSHOT in apache.snapshots (https://repository.apache.org/snapshots/)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:48)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:369)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder.lambda$wrap$0(RunnableErrorForwarder.java:73)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:627)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:262)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:520)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:408)
	... 18 common frames omitted
```

We need to set up a deploy action to always have the 3.21.0 in sync

**Comment by oscerd** [comment]
BTW we still have conflicts.

**Review by davsclaus — APPROVED** [review]

**Comment by ank19** [comment]
@oscerd - Hm, can you give me a hint how to detect the conflict? I get no indication that there is one...
image

**Comment by oscerd** [comment]
My fault, i didn't reload the PR page. It's good to go now.

**Comment by oscerd** [comment]
Can you also backport to main? Thanks.

**Comment by ank19** [comment]
@oscerd - Sure, just created a PR: https://github.com/apache/camel/pull/9577

**Pull request merged**